standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/StatisticsManagementTask.java

```diff
@@ -0,0 +1,201 @@
+ * metastore task runs periodically, it will drop partitions with age (creation time) greater than retention period.
+ * Dropping partitions after retention period will also delete the data in that partition.
+ *
+ */
+public class StatisticsManagementTask implements MetastoreTaskThread {
+    private static final Logger LOG = LoggerFactory.getLogger(StatisticsManagementTask.class);
+
+    // global
+    public static final String STATISTICS_AUTO_DELETION = "statistics.auto.deletion";
+    public static final String STATISTICS_RETENTION_PERIOD = "statistics.retention.period";
+
+    // The 2 configs for users to set in the conf
+    // this is an optional table property, if this property does not exist for a table, then it is not excluded
+    public static final String STATISTICS_AUTO_DELETION_EXCLUDE_TBLPROPERTY = "statistics.auto.deletion.exclude";
+
+    private static final Lock lock = new ReentrantLock();
+
+    // these are just for testing
+    private static int completedAttempts;
+    private static int skippedAttempts;
+
+    private Configuration conf;
+
+    @Override
+    public long runFrequency(TimeUnit unit) {
+        return MetastoreConf.getTimeVar(conf, MetastoreConf.ConfVars.STATISTICS_MANAGEMENT_TASK_FREQUENCY, unit);
+    }
+
+    @Override
+    public void setConf(Configuration configuration) {
+        // we modify conf in setupConf(), so we make a copy
+        conf = new Configuration(configuration);
+    }
+
+    @Override
+    public Configuration getConf() {
+        return conf;
+    }
+
+    // what needs to be included in this run() method
+    // find all the tables need to be deleted
+    // delete all column stats
+    @Override
+    public void run() {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Auto statistics deletion started. Cleaning up table/partition column statistics over the retention period.");
+        }
+        if (!MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.STATISTICS_AUTO_DELETION)) {
+            LOG.info("Statistics auto deletion is set to off currently.");
+            return;
+        }
+        if (lock.tryLock()) {
+            skippedAttempts = 0;
+            String qualifiedTableName = null;
+            IMetaStoreClient msc = null;
+            try {
+                // Get retention period in conf in milliseconds; default is 365 days.
+                long retentionMillis = 24 * 60 * 60 * 1000 * MetastoreConf.getIntVar(conf, MetastoreConf.ConfVars.STATISTICS_RETENTION_PERIOD);
+                long now = System.currentTimeMillis();
+
+                // Get all databases from metastore
+                List<String> databases = msc.getAllDatabases();
+                for (String dbName : databases) {
+                    // Get all tables in the database
+                    List<String> tableNames = msc.getAllTables(dbName);
+                    for (String tableName : tableNames) {
```

Comment:
No need to scan through the whole tables, there is a `LAST_ANALYZED` in the table `TAB_COL_STATS` or `PART_COL_STATS`, use this field against the `retentionMillis` to select the statistics that need to drop